From b1e4476cc603eaa1bda7a799842f79ecd4f8961a Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Mon, 19 Mar 2012 11:54:12 +0100
Subject: [PATCH] Fix input device for M-audio fasttrack pro

Some M-audio fasttrack pro devices, the input device is at index 1 instead of index 0.
According to
http://mailman.alsa-project.org/pipermail/alsa-devel/2012-March/050701.html
the reason for this is probably that the device has mutually exclusive
analog and digital input. With this patch we can catch the input regardless
of state.

BugLink: https://bugs.launchpad.net/bugs/569932
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 src/modules/alsa/mixer/profile-sets/maudio-fasttrack-pro.conf | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/modules/alsa/mixer/profile-sets/maudio-fasttrack-pro.conf b/src/modules/alsa/mixer/profile-sets/maudio-fasttrack-pro.conf
index 75f5112..c8025fd 100644
--- a/src/modules/alsa/mixer/profile-sets/maudio-fasttrack-pro.conf
+++ b/src/modules/alsa/mixer/profile-sets/maudio-fasttrack-pro.conf
@@ -33,9 +33,11 @@ device-strings = hw:%f,0,0
 channel-map = left,right
 direction = output
 
+; Try both device 0 and device 1 for input, see
+; http://mailman.alsa-project.org/pipermail/alsa-devel/2012-March/050701.html
 [Mapping analog-stereo-a-input]
 description = Analog Stereo Channel A
-device-strings = hw:%f,0,0
+device-strings = hw:%f,0,0 hw:%f,1,0
 channel-map = left,right
 direction = input
 
-- 
1.9.3

